From 6e7681cb7347c10a99645246803b1170575694b2 Mon Sep 17 00:00:00 2001
From: Christian Hemp <c.hemp@phytec.de>
Date: Sat, 31 Aug 2013 23:54:14 +0200
Subject: [PATCH 15/55] pfla02: Add partitions for sd card

Add barebox and bareboxenv partitions to sd.
Also add env and self for SD card.

Signed-off-by: Christian Hemp <c.hemp@phytec.de>
---
 arch/arm/boards/phytec-pfla02/board.c   |   15 ++++++++++-----
 arch/arm/dts/imx6qdl-phytec-pbab01.dtsi |    9 ++++++++-
 arch/arm/dts/imx6qdl-phytec-pfla02.dtsi |   12 ++++++++++++
 3 files changed, 30 insertions(+), 6 deletions(-)

diff --git a/arch/arm/boards/phytec-pfla02/board.c b/arch/arm/boards/phytec-pfla02/board.c
index a6ce633..c56d7a4 100644
--- a/arch/arm/boards/phytec-pfla02/board.c
+++ b/arch/arm/boards/phytec-pfla02/board.c
@@ -112,7 +112,7 @@ static int phytec_pfla02_init(void)
 				flash_header_end - flash_header_start, 0);
 
 
-	imx6_bbu_internal_mmc_register_handler("mmc", "/dev/mmc2",
+	imx6_bbu_internal_mmc_register_handler("mmc", "/dev/mmc2.barebox",
 			0, (void *)flash_header_start, flash_header_end - flash_header_start, 0);
 
 	eth_phy_reset();
@@ -130,14 +130,19 @@ static int phytec_pfla02_init(void)
 		envdev = "NAND";
 		break;
         case BOOTSOURCE_MMC:
+		of_device_enable_path("/chosen/environment-mmc");
+		devfs_add_partition("mmc2", 0, SZ_512K, 
+					DEVFS_PARTITION_FIXED, "self0");
+		devfs_add_partition("mmc2", SZ_512K, SZ_64K,
+					DEVFS_PARTITION_FIXED, "env0");
 		envdev = "MMC";
                 break;
         default:
         case BOOTSOURCE_SPI:
-                of_device_enable_path("/chosen/environment");
-                devfs_add_partition("m25p0", 0, SZ_512K, DEVFS_PARTITION_FIXED, "self0");
-                devfs_add_partition("m25p0", SZ_512K, SZ_64K,
-                                    DEVFS_PARTITION_FIXED, "env0");
+                of_device_enable_path("/chosen/environment-spi");
+		devfs_add_partition("m25p0", 0, SZ_512K, DEVFS_PARTITION_FIXED, "self0");
+		devfs_add_partition("m25p0", SZ_512K, SZ_64K,
+			DEVFS_PARTITION_FIXED, "env0");
 		envdev = "SPI-NOR";
                 break;
         }
diff --git a/arch/arm/dts/imx6qdl-phytec-pbab01.dtsi b/arch/arm/dts/imx6qdl-phytec-pbab01.dtsi
index ed43d15..288ad61 100644
--- a/arch/arm/dts/imx6qdl-phytec-pbab01.dtsi
+++ b/arch/arm/dts/imx6qdl-phytec-pbab01.dtsi
@@ -11,9 +11,16 @@
 
 / {
 	chosen {
-		environment@0 {
+		environment-spi {
 			compatible = "barebox,environment";
 			device-path = &flash, "partname:barebox-environment";
+			status = "disabled";
+		};
+
+		enviroment-mmc {
+			compatible = "barebox,environment";
+			device-path = &usdhc3, "partname:barebox-environment";
+			status = "disabled";
 		};
 	};
 };
diff --git a/arch/arm/dts/imx6qdl-phytec-pfla02.dtsi b/arch/arm/dts/imx6qdl-phytec-pfla02.dtsi
index e800e8f..ff9c6c8 100644
--- a/arch/arm/dts/imx6qdl-phytec-pfla02.dtsi
+++ b/arch/arm/dts/imx6qdl-phytec-pfla02.dtsi
@@ -59,4 +59,16 @@
         cd-gpios = <&gpio1 27 0>;
         wp-gpios = <&gpio1 29 0>;
         status = "disabled";
+	#address-cells = <1>;
+	#size-cells = <1>;
+	
+	partition@0 {
+		label = "barebox";
+		reg = <0x0 0x80000>;
+	};
+
+	partition@1 {
+		label = "barebox-environment";
+		reg = <0x80000 0x10000>;
+	};
 };
-- 
1.7.0.4

